src/manifest/store.rs

```diff
@@ -486,55 +487,62 @@ impl StoredManifest {
     }
 }
 
-/// Represents the metadata of a manifest file stored in the object store.
+/// Represents the metadata of a manifest stored in the backend.
 #[derive(Serialize, Debug)]
-pub(crate) struct ManifestFileMetadata {
+pub(crate) struct ManifestMetadata {
     pub(crate) id: u64,
-    #[serde(serialize_with = "serialize_path")]
-    pub(crate) location: Path,
     pub(crate) last_modified: chrono::DateTime<Utc>,
     #[allow(dead_code)]
     pub(crate) size: u32,
 }
 
-fn serialize_path<S>(path: &Path, serializer: S) -> Result<S::Ok, S::Error>
-where
-    S: serde::Serializer,
-{
-    serializer.serialize_str(path.as_ref())
+pub(crate) struct ManifestStore {
+    backend: Box<dyn ManifestStoreBackend>,
+    clock: Arc<dyn Clock + Send + Sync>,
 }
 
-pub(crate) struct ManifestStore {
+#[async_trait]
+pub trait ManifestStoreBackend: Send + Sync {
+    async fn write_manifest(&self, id: u64, manifest: &Manifest) -> Result<(), SlateDBError>;
```

Comment:
not: shouldn't it be `write_manifest_if_not_exists`?

Reply:
sounds good 👍